merkledag/coding.go

```diff
@@ -92,7 +93,10 @@ func (n *Node) Encoded(force bool) ([]byte, error) {
 
 // Decoded decodes raw data and returns a new Node instance.
 func Decoded(encoded []byte) (*Node, error) {
-	n := &Node{}
+	n := new(Node)
 	err := n.Unmarshal(encoded)
-	return n, err
+	if err != nil {
+		return nil, errors.New("incorrectly formatted merkledag node")
```

Comment:
I thought about that, but this is an error that will be passed to the user directly (in most cases). Whatever error Unmarshal spits out is not going to be something we want the users to see